src/utils/validations.js

```diff
@@ -1 +1,22 @@
+import { DNS1123_START_ERROR, DNS1123_END_ERROR, DNS1123_CONTAINS_ERROR, DNS1123_EMPTY_ERROR } from './strings';
+
 export const isPositiveNumber = value => value && value.match(/^[1-9]\d*$/);
+
+// DNS-1123 subdomain
+export const validateDNS1123SubdomainValue = value => {
+  if (!value) {
+    return DNS1123_EMPTY_ERROR;
+  }
+  if (!value.charAt(0).match('[a-zA-Z0-9]')) {
```

Comment:
Looking at implementation of k8s DNS1123 validation, upper case characters are not allowed.
I think it's ok for VM name let's write a comment if we reference 1123 in the function name.

Reply:
backend will actually convert the name to lowercase, maybe I could show warning that such conversion will happen ? WDYT ?

Reply:
Warning would be great if it appears once the user enters upper case letter.

Reply:
> backend will actually convert the name to lowercase, maybe I could show warning that such conversion will happen ? WDYT ?

I feel like most UI users either won't see these names of the backend or will understand that it needed to be changed. Just wondering if a warning would be concerning to these folks. I'm trying to think of something subtle, but don't have any good ideas yet :)

Reply:
They will see `myvirtualmachine` instead of `MyVirtualMachine`  in VM List for example. Maybe just info instead of warning ? Or nothing, im fine with either. Its your call @lizsurette

Reply:
@rawagner Ah sorry I misunderstood then :) Yes, definitely makes sense to highlight this to the user as to what's happening since it will change how it's represented in the UI. Info makes sense as a level of notification to me rather than warning.

Reply:
Latest PR update adds possibility to specify the type of validation message (error/warning/info). 

info message about uppercase to lowercase conversion
![validation](https://user-images.githubusercontent.com/2078045/48696396-1e66f400-ebe2-11e8-990f-55488653e050.png)

error
![validation1](https://user-images.githubusercontent.com/2078045/48696427-2f176a00-ebe2-11e8-9259-afab11ddbb8b.png)


and I do not have warning example since we do not have any warnings in this case :) just error/info

Reply:
The info message and behavior look good to me. I think adding `text-transform: lowercase;` to the `input` so that it behaves like [this](https://codepen.io/abraren/pen/MzxRJL?editors=1100) would be a nice future enhancement. If this is added, the info message would probably change to "Name must be lowercase".

Reply:
followup issue https://github.com/kubevirt/web-ui-components/issues/129